# Analysing the ACIS Background with the "Blank-Sky" Files

In [1]:
#source /export/CIAOT/ciaot_install/20180306/ciao-4.10/bin/ciao.bash -o
PFILES="./param;$ASCDS_INSTALL/param:$ASCDS_INSTALL/contrib/param"
mkdir -p ./param

In [2]:
/bin/rm -rf 1838 
download_chandra_obsid 1838 evt1,pbk,bias,dtf,msk,mtl,flt,stat,asol,fov,bpix


  Type     Format      Size  0........H.........1  Download Time Average Rate
  ---------------------------------------------------------------------------
  bpix     fits       12 Kb  ####################          < 1 s  1678.7 kb/s
  fov      fits        6 Kb  ####################          < 1 s  857.6 kb/s
  asol     fits        2 Mb  ####################          < 1 s  56170.5 kb/s
  evt1     fits       15 Mb  ####################          < 1 s  89770.3 kb/s
  flt      fits        7 Kb  ####################          < 1 s  959.4 kb/s
  msk      fits        5 Kb  ####################          < 1 s  792.3 kb/s
  mtl      fits      341 Kb  ####################          < 1 s  22088.8 kb/s
  stat     fits      288 Kb  ####################          < 1 s  19374.5 kb/s
  bias     fits      420 Kb  ####################          < 1 s  25006.5 kb/s
  bias     fits      493 Kb  ####################          < 1 s  23289.8 kb/s
  bias     fits      426 Kb  ####################          <

In [3]:
chandra_repro 1838 out= 


Processing input directory '/stage/pool1/kjg/Nb/1838'

Resetting afterglow status bits in evt1.fits file...

Running acis_build_badpix and acis_find_afterglow to create a new bad pixel file...

Running acis_process_events to reprocess the evt1.fits file...
Filtering the evt1.fits file by grade and status and time...
Applying the good time intervals from the flt1.fits file...
The new evt2.fits file is: /stage/pool1/kjg/Nb/1838/repro/acisf01838_repro_evt2.fits

Updating the event file header with chandra_repro HISTORY record
Creating FOV file...
Setting observation-specific bad pixel file in local ardlib.par.

Cleaning up intermediate files

         /stage/pool1/kjg/Nb/1838/repro/acisf01838_repro_bpix1.fits
         Run 'punlearn ardlib' when analysis of this dataset completed.

The data have been reprocessed.
Start your analysis with the new products in
/stage/pool1/kjg/Nb/1838/repro



In [4]:
mv 1838/repro/* .

## Single Chip

In [5]:
dmcopy "acisf01838_repro_evt2.fits[energy=500:7000,ccd_id=7]" 1838_c7.fits cl+

## Remove background flare


In [6]:
cat << EOM > sources.reg
# Region file format: CIAO version 1.0
circle(4072.5,4252.5,400)
circle(4604.5,4804.5,48)
EOM

dmcopy "1838_c7.fits[exclude sky=region(sources.reg)]" 1838_c7_bg.fits cl+

In [7]:
dmextract infile="1838_c7_bg.fits[bin time=84245789.155:84253741.155:259.28]" \
  outfile=1838_c7_bg.lc op=ltc1 clob+

In [8]:
deflare 1838_c7_bg.lc 1838_c7_bg_deflare.gti method=clean

Parameters used to clean the lightcurve are:
  script version = May 2017
  mean           = None
  clip           = 3
  scale          = 1.2
  minfrac        = 0.1
  outfile        = 1838_c7_bg_deflare.gti
  plot           = False

Total number of bins in lightcurve   = 31
Max length of one bin                = 255.997 s
Num. bins with a smaller exp. time   = 1
Num. bins with exp. time = 0         = 0
Calculated an initial mean (sigma-clipped) rate of 0.269603 ct/s
Lightcurve limits use a scale factor of 1.2 about this mean
Filtering lightcurve between rates of 0.224669 and 0.323524 ct/s
Number of good time bins = 24
Rate filter:  0.22466912825964613 <= count_rate < 0.32352354469389039 
Mean level of filtered lightcurve = 0.26441449970405073 ct/s

GTI limits calculated using a count-rate filter:
  (count_rate>0.22466912825964613 && count_rate<0.32352354469389039)

The corresponding times are:
  ((time >= 84246048.435000002) && (time < 84247604.11500001)) ; 1.54 ksec, bin 1
  ((time >= 

In [9]:
dmcopy "1838_c7.fits[@1838_c7_bg_deflare.gti]" 1838_c7_clean.fits cl+

## Now back to thread

In [10]:
ls -1 $CALDB/data/chandra/acis/bkgrnd/

acis0D2000-12-01bgstow_ctiN0005.fits
acis0D2005-09-01bgstow_ctiN0002.fits
acis0D2009-09-21bgstow_ctiN0002.fits
acis0iD1999-09-16bkgrndN0003.fits
acis0iD2000-01-29bkgrndN0004.fits
acis0iD2000-01-29bkgrnd_ctiN0006.fits
acis0iD2000-12-01bkgrndN0003.fits
acis0iD2000-12-01bkgrnd_ctiN0005.fits
acis0iD2005-09-01bkgrnd_ctiN0005.fits
acis0iD2009-09-21bkgrnd_ctiN0003.fits
acis0iD2012-01-01bkgrnd_ctiN0002.fits
acis1D2000-12-01bgstow_ctiN0005.fits
acis1D2005-09-01bgstow_ctiN0002.fits
acis1D2009-09-21bgstow_ctiN0002.fits
acis1iD1999-09-16bkgrndN0003.fits
acis1iD2000-01-29bkgrndN0004.fits
acis1iD2000-01-29bkgrnd_ctiN0006.fits
acis1iD2000-12-01bkgrndN0003.fits
acis1iD2000-12-01bkgrnd_ctiN0005.fits
acis1iD2005-09-01bkgrnd_ctiN0005.fits
acis1iD2009-09-21bkgrnd_ctiN0003.fits
acis1iD2012-01-01bkgrnd_ctiN0002.fits
acis1sD2000-01-29bkgrndN0006.fits
acis1sD2000-01-29bkgrnd_ctiN0006.fits
acis1sD2000-12-01bkgrndN0002.fits
acis1sD2000-12-01bkgrnd_ctiN0006.fits
acis2D2000-12-01bgstow_ctiN0003.fits
acis2D2005-09

In [11]:
acis_bkgrnd_lookup 1838_c7_clean.fits

/export/CIAOT/ciaot_install/20181129/ciao-4.11/CALDB/data/chandra/acis/bkgrnd/acis7sD2000-01-29bkgrnd_ctiN0002.fits


In [12]:
 pget acis_bkgrnd_lookup outfile

/export/CIAOT/ciaot_install/20181129/ciao-4.11/CALDB/data/chandra/acis/bkgrnd/acis7sD2000-01-29bkgrnd_ctiN0002.fits


In [13]:
acis_bkgrnd_lookup "acisf01838_repro_evt2.fits[sky=circle(6072.5,4632.5,320)]"

/export/CIAOT/ciaot_install/20181129/ciao-4.11/CALDB/data/chandra/acis/bkgrnd/acis1sD2000-01-29bkgrnd_ctiN0006.fits
/export/CIAOT/ciaot_install/20181129/ciao-4.11/CALDB/data/chandra/acis/bkgrnd/acis3sD2000-01-29bkgrnd_ctiN0006.fits


In [14]:
pget acis_bkgrnd_lookup outfile

/export/CIAOT/ciaot_install/20181129/ciao-4.11/CALDB/data/chandra/acis/bkgrnd/acis1sD2000-01-29bkgrnd_ctiN0006.fits,/export/CIAOT/ciaot_install/20181129/ciao-4.11/CALDB/data/chandra/acis/bkgrnd/acis3sD2000-01-29bkgrnd_ctiN0006.fits


## Make a local copy


In [15]:
cp -fv `acis_bkgrnd_lookup 1838_c7_clean.fits` \
      bgevt2_c7.fits


`/export/CIAOT/ciaot_install/20181129/ciao-4.11/CALDB/data/chandra/acis/bkgrnd/acis7sD2000-01-29bkgrnd_ctiN0002.fits' -> `bgevt2_c7.fits'


## Tailor background


In [16]:
dmkeypar 1838_c7_clean.fits DATAMODE echo+

FAINT


In [17]:
dmkeypar 1838_c7_clean.fits GAINFILE echo+

acisD2000-01-29gain_ctiN0007.fits


In [18]:
dmkeypar bgevt2_c7.fits GAINFILE echo+

acisD2000-01-29gain_ctiN0006.fits


## Reapply gain

In [19]:
acis_process_events infile=bgevt2_c7.fits outfile=bgevt2_c7_newgain.fits \
      acaofffile=NONE stop="none" doevtgrade=no apply_cti=yes apply_tgain=no \
      calculate_pi=yes pix_adj=NONE \
      gainfile=$CALDB/data/chandra/acis/det_gain/acisD2000-01-29gain_ctiN0007.fits \
      eventdef="{s:ccd_id,s:node_id,i:expno,s:chip,s:tdet,f:det,f:sky,s:phas,l:pha,l:pha_ro,f:energy,l:pi,s:fltgrade,s:grade,x:status}"


# acis_process_events (CIAO 4.11): WARNING: The CTI adjustment to PHA will not be (re)computed or removed because doevtgrade=no.
# acis_process_events (CIAO 4.11): WARNING: acaofffile is NONE and OBS_MODE is pointing

# acis_process_events (CIAO 4.11): WARNING: The values of PHA, ENERGY, PI, FLTGRADE and GRADE may be inaccurate because doevtgrade=no.
# acis_process_events (CIAO 4.11): WARNING: problem reading ctifile, cti adjustment will not be applied. Changing apply_cti=yes to apply_cti=no.
# acis_process_events (CIAO 4.11): WARNING: The eventdef specifies the tdet column although it is not in the input file and the requested coordinate transforms do not calculate it.


In [20]:
dmkeypar bgevt2_c7_newgain.fits GAINFILE echo+

acisD2000-01-29gain_ctiN0007.fits


## ~~Adding PNT header keywords~~

In CIAO 4.11 `reproject_events` will now copy these keywords itself.

In [21]:
### dmlist bgevt2_c7.fits header |grep PNT

In [22]:
### dmmakepar 1838_c7_clean.fits event_header.par cl+

In [23]:
### grep _pnt event_header.par > event_pnt.par
### grep _avg event_header.par >> event_pnt.par
### cat event_pnt.par

In [24]:
### chmod +w bgevt2_c7.fits
### dmreadpar event_pnt.par "bgevt2_c7.fits[events]" clobber+
### dmlist bgevt2_c7_newgain.fits header |grep PNT

## Reproject background data

In [25]:
punlearn reproject_events
pset reproject_events infile=bgevt2_c7_newgain.fits
pset reproject_events outfile=bgevt2_c7_reproj.fits
pset reproject_events aspect=pcadf084244404N003_asol1.fits
pset reproject_events match=1838_c7_clean.fits
pset reproject_events random=0
reproject_events mode=h cl+


In [26]:
dmlist bgevt2_c7_reproj.fits header,clean | egrep '_NOM|_PNT|_AVG|SIM_'

SIM_X                   -0.68282252473119           SIM focus pos (mm)
SIM_Y                                   0           SIM orthogonal axis pos (mm)
SIM_Z                     -187.2058600929           SIM translation stage pos (mm)
RA_NOM                     278.3865720663           Nominal RA
DEC_NOM                    -10.5892699157           Nominal Dec
ROLL_NOM                                0 [deg]     Roll (fake)
RA_PNT                     278.3865720663 [deg]     Pointing RA
DEC_PNT                    -10.5892699157 [deg]     Pointing Dec
ROLL_PNT                   266.8075204601 [deg]     Pointing Roll
DY_AVG                   0.0940931961060 [mm]       Mean DY during observation
DZ_AVG                   0.239958904840 [mm]        Mean DZ during observation
DTH_AVG                 -0.00477186496940 [deg]     Mean DTHETA during observation
CBD40001             SIM_Z(-206.8000030518--174.1999969482)mm ACIS-S in focal plane


## Imaging Analysis


In [27]:
fluximage 1838_c7_clean.fits out=1838_c7 bin=8 band=broad clob+

Running fluximage
Version: 12 September 2016

Using CSC ACIS broad science energy band.
Aspect solution pcadf084244404N003_asol1.fits found.
Bad-pixel file acisf01838_repro_bpix1.fits found.
Mask file acisf01838_000N003_msk1.fits found.

The output images will have 140 by 140 pixels, pixel size of 3.936 arcsec,
    and cover x=3632.5:4752.5:8,y=3840.5:4960.5:8.

Running tasks in parallel with 4 processors.
Creating aspect histogram for obsid 1838
Creating instrument map for obsid 1838
Creating exposure map for obsid 1838
Thresholding data for obsid 1838
Exposure-correcting image for obsid 1838

The following files were created:

 The clipped counts image is:
     1838_c7_broad_thresh.img

 The clipped exposure map is:
     1838_c7_broad_thresh.expmap

 The exposure-corrected image is:
     1838_c7_broad_flux.img



In [28]:
get_sky_limits 1838_c7_broad_thresh.img
dmf=`pget get_sky_limits dmfilter`

Running: get_sky_limits
  version: 07 October 2016
Checking binning of image: 1838_c7_broad_thresh.img
  Image has 140 x 140 pixels
  Pixel size is 8.0 by 8.0
  Lower left (0.5,0.5) corner is x,y= 3632.5, 3840.5
  Upper right (140.5,140.5) corner is x,y= 4752.5, 4960.5
  DM filter is:
    x=3632.5:4752.5:#140,y=3840.5:4960.5:#140
  mkexpmap xygrid value is:
    3632.5:4752.5:#140,3840.5:4960.5:#140


In [29]:
dmcopy "bgevt2_c7_reproj.fits[energy=500:7000][bin $dmf]" 1838_c7_broad.bgimg cl+

### Compute scaling factor

In [30]:
bkg_cts=`dmlist bgevt2_c7_reproj.fits"[energy=9000:12000]" counts`
bkg_exp=`dmkeypar bgevt2_c7_reproj.fits LIVETIME echo+`

bkg_rate=`echo $bkg_cts/$bkg_exp | bc -l`
echo $bkg_cts
echo $bkg_exp
echo $bkg_rate

85807
110000.0
.78006363636363636363


In [31]:
obs_cts=`dmcopy "acisf01838_repro_evt2.fits[ccd_id=7][@1838_c7_bg_deflare.gti]" - | \
  dmlist "-[energy=9000:12000]" counts `
obs_exp=`dmkeypar "acisf01838_repro_evt2.fits[ccd_id=7][@1838_c7_bg_deflare.gti]" \
  LIVETIME echo+`

obs_rate=`echo $obs_cts/$obs_exp | bc -l`
echo $obs_cts
echo $obs_exp
echo $obs_rate


4788
6060.75475111337
.79000061817720590191


In [32]:
dmimgcalc 1838_c7_broad_thresh.img,1838_c7_broad_thresh.expmap,1838_c7_broad.bgimg \
  none 1838_c7_broad.netimg \
  op="imgout=((img1-(((0.0+${obs_cts})/${bkg_cts})*img3))/img2)" cl+

BTIMCORR not present in all input files...FAIL...
BTIMDRFT not present in all input files...FAIL...
BTIMNULL not present in all input files...FAIL...
BTIMRATE not present in all input files...FAIL...
omit - FOC_LEN values different more than 1.000000
OBI_NUM not present in all input files...FAIL...
omit - RAND_SKY values different more than 0.050000
omit - ROLL_NOM values different more than 1.000000


In [33]:
### ds9 1838_c7_broad.netimg

# Using `blanksky` and `blansky_image`

In [34]:
pset blanksky evt="acisf01838_repro_evt2.fits[ccd_id=7][@1838_c7_bg_deflare.gti]"
pset blanksky out=1838_blank.evt
pset blanksky tmpdir=./
blanksky mode=h clob+

Running blanksky
Version: 18 October 2018

Aspect solution file pcadf084244404N003_asol1.fits found.

ACIS background file /export/CIAOT/ciaot_install/20181129/ciao-4.11/CALDB/data/chandra/acis/bkgrnd/acis7sD2000-01-29bkgrnd_ctiN0002.fits found.

Calculated scale factor for 'weight_method=particle':
    ACIS-7: 0.05579314 written to the 'BKGSCAL7' header keyword.


In [35]:
pset blanksky_image bkg=1838_blank.evt
pset blanksky_image out=1838_blank
pset blanksky_image img=1838_c7_broad_thresh.img
pset blanksky_image tmp=./
blanksky_image mode=h clob+ verb=1

Running blanksky_image
Version: 8 August 2017



In [36]:
### ds9 1838_blank_particle_bkgsub.img

# Cleanup

In [37]:
/bin/rm -rf 1838
/bin/rm -f acisf01838_*.fits
/bin/rm -f pcadf*_asol1.fits acisf01838_asol1.lis